diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 00000000..94824273
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,10 @@
+module(
+    name = "seal",
+    version = "4.1.1",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "zlib", version = "1.2.13")
+bazel_dep(name = "microsoft_gsl", version = "4.0.0")
+bazel_dep(name = "zstd", version = "1.5.6")
+bazel_dep(name = "rules_foreign_cc", version = "0.12.0")